ducktape/tests/test.py

```diff
@@ -35,27 +35,57 @@ def __init__(self, test_context, *args, **kwargs):
         self.test_context = test_context
         self.logger = test_context.logger
 
+    def who_am_i(self):
+        """Human-readable name for help with logging."""
+        return self.__module__.split(".")[-1] + "." + self.__class__.__name__
+
     def min_cluster_size(self):
-        """
-        Provides a helpful heuristic for determining if there are enough nodes in the cluster to run this test.
+        """Heuristic for guessing whether there are enough nodes in the cluster to run this test.
 
         Note this is not a reliable indicator of the true minimum cluster size, since new service instances may
         be added at any time. However, it does provide a lower bound on the minimum cluster size.
         """
         return self.test_context.services.num_nodes()
 
+    def setUp(self):
+        """Override this for custom setup logic."""
+        pass
+
     def tearDown(self):
         """Default teardown method which stops and cleans services registered in the ServiceRegistry.
         Note that there is not a default setUp method. This is because the framework makes no assumptions
         about when and in what methods the various services are started.
+
+        Catch all exceptions so that we try to do the full clean and free process.
+        Propagate KeyboardInterrupt so the top level test runner can decide how to handle it.
         """
         if hasattr(self.test_context, 'services') and type(self.test_context.services) == ServiceRegistry:
-            self.test_context.services.stop_all()
-            self.copy_service_logs()
-            self.test_context.services.clean_all()
+            exceptions = []
+            try:
+                self.test_context.services.stop_all()
+            except BaseException as e:
+                exceptions.append(e)
```

Comment:
Isn't it probably bad if we're seeing exceptions here? Right now they're just being swallowed. It seems like we should at least log them?